### DISCUSSION TITLE
Toggling input parsing/processing

### DISCUSSION THREAD

**Discussion by A1-exe** [python-cmd2/cmd2#1142] [Q&A] [answered]
Is it possible to temporarily disable cmd2's input processing? Maybe function calls to enable/disable the handler? 
For example, when I create a command and use input() to do things, it still shows up in the history for cmd2.

**Comment by kmvanbrunt** [comment, marked as answer]
GNU readline adds to its up-arrow history by default when you call `input()`. For this reason, cmd2 has a wrapper method called `read_input()`.

This function provides a lot of more features than `input()`, like custom tab completion and custom up-arrow history items while the user is typing their input. For the case in which you want no history available to the user while typing and you do not want to store their input in the history, just call like this: `self.read_input(prompt)`.

Here is the function signature.
https://cmd2.readthedocs.io/en/latest/api/cmd.html#cmd2.Cmd.read_input

Here is an example that demonstrates the various ways you can use `read_input()`.
https://github.com/python-cmd2/cmd2/blob/master/examples/read_input.py
